Remove build files

JDBetteridge opened the pull request:
This addresses issue #1771 .
I have also bumped the VTK version forward to get the latest wheels available on PyPI.

JDBetteridge commented:
I've changed the Docker function in the Jenkins file, but this is not tested, can someone double check I haven't done anything stupid here

wence- commented:
> I've changed the Docker function in the Jenkins file, but this is not tested, can someone double check I haven't done anything stupid here

Is your intention to `--remove-build-files` inside the docker images? That seems like a reasonable thing to aim for. If so, you have not done that.

You can test that the docker build works by changing the `if branch == master` line in the Jenkinsfile (don't forget to remove the lines that push the new images!)

JDBetteridge commented:
> > I've changed the Docker function in the Jenkins file, but this is not tested, can someone double check I haven't done anything stupid here
> 
> Is your intention to `--remove-build-files` inside the docker images? That seems like a reasonable thing to aim for. If so, you have not done that.
> 
> You can test that the docker build works by changing the `if branch == master` line in the Jenkinsfile (don't forget to remove the lines that push the new images!)

Ah I had assumed that Docker was using the Jenkins build, when it clearly isn't! Changed now, let's see if it passes.

dham commented on firedrake/paraview_reordering.py:
```diff
@@ -1,18 +1,36 @@
-However, we only need vtkCommonKitPython, which, according to ldd, only cares about
-things that we should expect: libc, libdl.so, libstdc++, libm, libgcc_s.
-Thus, we hackily import the module that lives in vtkCommonKitPython.so
+This requires (even more) explanation.
```
The reader won't know what "even more" refers to. Just drop it.

dham commented on docker/Dockerfile.vanilla:
```diff
@@ -9,5 +9,5 @@ USER firedrake
-RUN curl -O https://raw.githubusercontent.com/firedrakeproject/firedrake/master/scripts/firedrake-install
-RUN bash -c "python3 firedrake-install --no-package-manager --disable-ssh"
+RUN curl -O https://raw.githubusercontent.com/firedrakeproject/firedrake/remove-build-files/scripts/firedrake-install
+RUN bash -c "python3 firedrake-install --no-package-manager --disable-ssh --remove-build-files"
```
I presume we intend to leave this in here. Having small docker containers sounds like a Good Thing (and updating petsc in a docker container sounds odd).
